src/test/java/org/folio/permstest/RestVerticleTest.java

```diff
@@ -401,6 +401,31 @@ public void testDeletePermsUsersByIdDoesNotExist(TestContext context) {
     context.assertEquals("No permissions user found with id " + uuid, response.body.getString("text"));
   }
 
+  private Response deleteByUserId(String userId, TestContext context) {
+    return send(HttpMethod.DELETE, "/perms/users/" + userId + "?indexField=userId", null, context);
+  }
+
+  @Test
+  public void testDeletePermsUsersByUserIdDoesNotExist(TestContext context) {
+    var userId = UUID.randomUUID().toString();
+    var response = deleteByUserId(userId, context);
+    assertThat(response.code, is(404));
+    assertThat(response.body.getString("text"), is("No permissions user found with userId " + userId));
+  }
+
+  @Test
+  public void testDeletePermsUsersByUserId(TestContext context) {
+    var id = UUID.randomUUID().toString();
+    var userId = UUID.randomUUID().toString();
+    var permsUsers = new JsonObject().put("id", id).put("userId", userId).encodePrettily();
+    var response = send(HttpMethod.POST, "/perms/users", permsUsers, context);
+    assertThat(response.code, is(201));
+    response = deleteByUserId(id, context);
```

Comment:
This seems a little confusing. The id is being used in two different ways. One to create the the test data, and then again to cause the 404. Maybe just create another variable called `userIdNotFound` or perhaps just hardcoding a second userid in this line to test the 404.

Reply:
Although I see you have a test dedicated to the 404 test so maybe I'm misunderstanding the purpose of this line.